Sources/Operators.swift

```diff
@@ -260,40 +260,38 @@ public func +(lhs: UIOffset, rhs: CGSize) -> CGSize {
 public protocol CGFloatConvertible {}
 
 
-public extension CGFloatConvertible {
+extension CGFloatConvertible {
     
     public var horizontalOffset: UIOffset {
-        return UIOffset(horizontal: CGFloat(self), vertical: 0.0)
+        return UIOffset(horizontal: self.asCGFloat, vertical: 0.0)
     }
     
     public var verticalOffset: UIOffset {
-        return UIOffset(horizontal: 0.0, vertical: CGFloat(self))
+        return UIOffset(horizontal: 0.0, vertical: self.asCGFloat)
     }
-    
+
+    internal var asCGFloat: CGFloat {
+        if let cgFloat = self as? CGFloat {
+            return cgFloat
+        } else if let double = self as? Double {
+            return CGFloat(double)
+        } else if let float = self as? Float {
+            return CGFloat(float)
+        } else if let int = self as? Int {
+            return CGFloat(int)
+        } else {
+            ErrorHandler.assertionFailure("Can not convert \(self) to CGFloat!")
+            return 0.0
+        }
+    }
+
 }
 
 
 // MARK: – CGFloat Extension
 
 
-extension CGFloat: CGFloatConvertible {
-    
-    public init(_ convertible: CGFloatConvertible) {
```

Comment:
This one was fun. In Xcode 10.2 (but not 10.1): all `CGFloat` initializers would end up calling through this one, including the initializers on L285, L287, and L289. This would cause an infinite recursion. Instead of a initializer, I'm now using a more-explicit `asCGFloat` property above.

Technically, this is a breaking change, since this initializer is public. That said, this initializer was only useful for code that was extending or heavily working with `CGFloatConvertible` (an implementation detail of this library). I'm willing to take the gamble that no one was doing that.